### ISSUE TITLE
Sigmoid  for lsgan

### ISSUE THREAD

**Issue by fanpengustc** [vanhuyz/CycleGAN-TensorFlow#61]
Why sigmoid is not used in Discriminator last layer for lsgan, I do not understand

**Comment by vanhuyz** [comment]
As far as I understand, `least square loss` is mainly used in regression model, and we don't need activation function (sigmoid/softmax) here. But in classification model, where `cross entropy loss` is mainly used, activation function works as converting output to probability of each class.

The maths behinds it is very complicated that I can't explain fully here but in practice just remember: least square → no sigmoid/softmax, cross entropy → need sigmoid/softmax.
